cmd/gb-vendor/vendor/stringset.go

```diff
@@ -0,0 +1,49 @@
+package vendor
+
+// union returns the union of a and b.
+func union(a, b map[string]bool) map[string]bool {
+	r := make(map[string]bool)
+	for k := range a {
+		r[k] = true
+	}
+	for k := range b {
+		r[k] = true
+	}
+	return r
+}
+
+// intersection returns the intersection of a and b.
+func intersection(a, b map[string]bool) map[string]bool {
+	r := make(map[string]bool)
+	for k := range a {
+		if b[k] {
+			r[k] = true
+		}
+	}
+	return r
+}
+
+// difference returns the symetric difference of a and b.
+func difference(a, b map[string]bool) map[string]bool {
+	r := make(map[string]bool)
+	for k := range a {
+		if !b[k] {
+			r[k] = true
+		}
+	}
+	for k := range b {
+		if !a[k] {
+			r[k] = true
+		}
+	}
+	return r
+}
+
+// contains returns true if a contains all the elements in s.
+func contains(a map[string]bool, s ...string) bool {
+	var r bool
+	for _, e := range s {
+		r = a[e]
+	}
+	return r
```

Comment:
Thanks for spotting my bug.

On Tue, Jun 2, 2015 at 5:42 PM, Dávid Kaya <notifications@github.com> wrote:

> In cmd/gb-vendor/vendor/stringset.go
> <https://github.com/constabulary/gb/pull/179#discussion_r31498727>:
>
> > +	}
> > +	for k := range b {
> > +		if !a[k] {
> > +			r[k] = true
> > +		}
> > +	}
> > +	return r
> > +}
> > +
> > +// contains returns true if a contains all the elements in s.
> > +func contains(a map[string]bool, s ...string) bool {
> > +	var r bool
> > +	for _, e := range s {
> > +		r = a[e]
> > +	}
> > +	return r
>
> This implementation is not correct, it will return whether the last string
> from s is in the set, not all of them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/pull/179/files#r31498727>.
>